File: Point.py

class Point:
    row = 0
    col = 0

    def __init__(self, row = 0, col = 0):
        self.row = row
        self.col = col

    def dist(self, p):
        return abs(self.col - p.col) + abs(self.row - p.row)

File: Point.go
package main

type Point struct {
	row , col int
}



func (p *Point) Dist(o Point) int {
	csum, rsum:= p.col - o.col, p.row - o.row
	if csum < 0 {
		csum = -csum
	}
	if rsum < 0 {
		rsum = -rsum
	}

	return rsum + csum
}
